presto-hive/src/test/java/io/prestosql/plugin/hive/parquet/TestTimestampReader.java

```diff
@@ -0,0 +1,118 @@
+import java.util.Optional;
+
+import static io.prestosql.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.prestosql.plugin.hive.HiveTestUtils.getHiveSession;
+import static io.prestosql.spi.type.TimestampType.TIMESTAMP_MILLIS;
+import static java.util.Collections.singletonList;
+import static org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorFactory.getStandardStructObjectInspector;
+import static org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory.javaLongObjectInspector;
+import static org.apache.parquet.column.ParquetProperties.WriterVersion.PARQUET_1_0;
+import static org.apache.parquet.hadoop.ParquetOutputFormat.WRITER_VERSION;
+import static org.apache.parquet.schema.MessageTypeParser.parseMessageType;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestTimestampReader
+{
+    @Test
+    public void testTimestampBackedByInt64()
+            throws Exception
+    {
+        MessageType parquetSchema = parseMessageType("message hive_timestamp { optional int64 test (TIMESTAMP_MILLIS); }");
+        ContiguousSet<Long> epochMillisValues = ContiguousSet.create(Range.closedOpen((long) -1_000, (long) 1_000), DiscreteDomain.longs());
+        ImmutableList.Builder<SqlTimestamp> timestamps = new ImmutableList.Builder<>();
+        for (long value : epochMillisValues) {
+            timestamps.add(SqlTimestamp.fromMillis(3, value));
+        }
+
+        List<ObjectInspector> objectInspectors = singletonList(javaLongObjectInspector);
+        List<String> columnNames = ImmutableList.of("test");
+
+        ConnectorSession session = getHiveSession(new HiveConfig());
+
+        try (ParquetTester.TempFile tempFile = new ParquetTester.TempFile("test", "parquet")) {
+            JobConf jobConf = new JobConf();
+            jobConf.setEnum(WRITER_VERSION, PARQUET_1_0);
+
+            ParquetTester.writeParquetColumn(
+                    jobConf,
+                    tempFile.getFile(),
+                    CompressionCodecName.SNAPPY,
+                    ParquetTester.createTableProperties(columnNames, objectInspectors),
+                    getStandardStructObjectInspector(columnNames, objectInspectors),
+                    new Iterator<?>[] {epochMillisValues.iterator()},
+                    Optional.of(parquetSchema),
+                    false);
+
+            Iterator<SqlTimestamp> expectedValues = timestamps.build().iterator();
+            try (ConnectorPageSource pageSource = FileFormat.PRESTO_PARQUET.createFileFormatReader(session, HDFS_ENVIRONMENT, tempFile.getFile(), columnNames, ImmutableList.of(TIMESTAMP_MILLIS))) {
+                // skip a page to exercise the decoder's skip() logic
+                Page firstPage = pageSource.getNextPage();
+
+                assertThat(firstPage.getPositionCount())
+                        .withFailMessage("Expected first page to have at least 1 row")
+                        .isGreaterThan(0);
```

Comment:
the shorter, intermediate version that you had
```
assertTrue(firstPage.getPositionCount() > 0, "Expected first page to have at least 1 row");
```

produces equally informative exception message (assuming page.positionCount is non-negative)